### PR TITLE
Make no history work with nested stacks

### PR DESCRIPTION
**Pull request by martijn00**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Fixes replacing root not working in a lot of situations.

### :arrow_heading_down: What is the current behavior?

The current navigation root is not replaced.

### :new: What is the new behavior (if this is a feature change)?

It is now correctly replaced.

### :boom: Does this PR introduce a breaking change?

No

### :bug: Recommendations for testing

Set `NoHistory = true` in one of the pages and run the playground sample.

### :memo: Links to relevant issues/docs
Fixes https://github.com/MvvmCross/MvvmCross/issues/2320

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [x] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Nuspec files were updated (when applicable)
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by Cheesebaron** [comment]
How to test this? Which cases does it fix?

**Comment by martijn00** [comment]
Updated description.

**Comment by Cheesebaron** [comment]
On Forms.iOS. If I set `NoHistory = true` on `ChildPage`. The back button is briefly visible. Looks very strange.

**Comment by martijn00** [comment]
@Cheesebaron  You've got to set `Animated = false` to prevent that. This is iOS + Forms behavior.

File: MvvmCross-Forms/MvvmCross.Forms/Views/MvxFormsPagePresenter.cs [hunk 1]
```diff
@@ -172,8 +172,9 @@ public virtual void RegisterAttributeTypes(Dictionary<Type, MvxPresentationAttri
                     if (attribute.WrapInNavigationPage && FormsApplication.MainPage is MvxNavigationPage currentPage)
                     {
                         if (attribute.NoHistory)
```
Comment (Cheesebaron) [inline MvvmCross-Forms/MvvmCross.Forms/Views/MvxFormsPagePresenter.cs]: This `if` pattern seems duplicated a lot of places in this class. Can we encapsulate this please?

**Pull request merged**